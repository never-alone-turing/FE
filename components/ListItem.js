import React from "react";
import { View, Text, StyleSheet, TouchableOpacity } from "react-native";

const ListItem = ({item}) => {
  return (
    <TouchableOpacity style={styles.listItem}>
      <View style={styles.listItemView}>
        <Text style={styles.listItemText}>{item.title}</Text>
      </View>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  listItem: {
    padding: 15,
    backgroundColor: '#f8f8f8',
    borderBottomWidth: 1,   
    borderColor: "#eee"
  },
  listItemView: {
    flexDirection: 'row'
  },
  listItemText: {
    fontSize: 18
  }
})

export default ListItem;