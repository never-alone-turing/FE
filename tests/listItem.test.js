import React from 'react'
import {create} from 'react-test-renderer'
import ListItem from '../components/ListItem'
import renderer from 'react-test-renderer'

const mockItem = {
  name: 'User',
  time: '7:00',
  window: '15 minutes'
}

  describe("<ListItem />", () => {

    it('renders correctly', () => {
      const tree = renderer.create(<ListItem
        item={mockItem}
        />).toJSON();
    expect(tree).toMatchSnapshot()
    })

  })
