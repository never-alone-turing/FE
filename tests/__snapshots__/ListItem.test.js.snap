// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should match the snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#f8f8f8",
      "borderBottomWidth": 1,
      "borderColor": "#eee",
      "opacity": 1,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
        }
      }
    />
  </View>
</View>
`;
